### ISSUE TITLE
error with Version: 0.1.2 (deque branch)

### ISSUE THREAD

**Issue by ijgnd** [lovac42/HoochieMama#5]
I was curious about the latest version from the deque branch that you made today (Version: 0.1.2 as of https://github.com/lovac42/HoochieMama/commit/41d6f3ab5e8ae55a12e2bc695c7c32fb7a35b453). I don't know if it was meant for usage. 

Thanks for incorporating great new features into this add-on and again thanks for all your other add-ons! Unfortunately since the last version I used an error has crept in: I get neither subdeck randomization nor sorting by CUSTOM_SORT. 

Maybe this report contains something useful for you. If not just close it without comment.

I tried with Anki 2.0.52. The only add-on I used was HoochieMama. I added some code to it so that for every review the following info is output on the console: deckID, date of LatestReview, the due date; the current interval, the number of overdue days are. I copy&pasted some of these lines under [1]. My settings are under [2].

[1]

```csv
did;LatestReview_Date;Due_Date;ivl_Days;overdue_Days
DeckA; 2017-11-19; 2018-12-05; 0381; 0000
DeckA; 2018-07-30; 2018-12-05; 0129; 0000
DeckB; 2018-06-28; 2018-12-05; 0161; 0000
DeckB; 2018-11-29; 2018-12-05; 0006; 0000
DeckB; 2018-12-03; 2018-12-05; 0003; 0000
DeckB; 2018-12-04; 2018-12-05; 0001; 0000
DeckC; 2016-03-30; 2018-12-05; 0981; 0000
DeckD; 2018-11-05; 2018-12-04; 0029; 0001
DeckE; 2016-04-18; 2018-12-05; 0961; 0000
DeckE; 2016-11-29; 2018-12-05; 0737; 0000
DeckE; 2016-11-29; 2018-12-04; 0735; 0001
DeckE; 2016-12-30; 2018-12-05; 0706; 0000
DeckE; 2016-12-30; 2018-12-05; 0706; 0000
DeckE; 2017-01-05; 2018-12-05; 0700; 0000
DeckE; 2017-01-06; 2018-12-04; 0697; 0001
DeckE; 2017-01-18; 2018-12-05; 0686; 0000
DeckE; 2017-03-31; 2018-12-04; 0613; 0001
DeckE; 2017-04-25; 2018-12-05; 0589; 0000
DeckE; 2017-07-21; 2018-12-04; 0501; 0001
DeckE; 2017-10-08; 2018-12-05; 0423; 0000
DeckE; 2017-10-10; 2018-12-04; 0420; 0001
DeckE; 2017-10-30; 2018-12-04; 0400; 0001
DeckE; 2018-02-05; 2018-11-28; 0296; 0007
DeckE; 2018-02-18; 2018-12-05; 0290; 0000
DeckE; 2018-07-13; 2018-12-05; 0145; 0000
DeckE; 2018-11-20; 2018-12-05; 0016; 0000
```

[2]

```python
#NOTE: DEQUE may have compatibility issues with other addons.
# Deque enables high-performance O(1) operations w/ a larger queue size
USE_DEQUE = True
IMPOSE_SUBDECK_LIMIT = True #Require DEQUE, uses custom queue builder


#Ensures even distribution of sub-decks by prioritizing today's due reviews first.
PRIORITIZE_TODAY = False


#Warning! Chaning default settings below may impact performance.
# CUSTOM_SORT = None #Default: Randomize review
CUSTOM_SORT = SHOW_YOUNG_FIRST
# CUSTOM_SORT = SHOW_MATURE_FIRST
# CUSTOM_SORT = SHOW_LOW_REPS_FIRST
# CUSTOM_SORT = SHOW_HIGH_REPS_FIRST


#Anki's default QUEUE_LIMIT of 50 does not work well for parent decks,
#so we changed it. Cheap windows tablets may need to lower this number.
QUEUE_LIMIT = 256 #for custom sort only
```

**Comment by lovac42** [comment]
set IMPOSE_SUBDECK_LIMIT to False, that's doesn't work with custom sort. Thanks I will fix that later.

**Comment by lovac42** [comment]
This should fix the issue with custom sorts, https://github.com/lovac42/HoochieMama/blob/chunking/src/hoochieMama.py

**Comment by ijgnd** [comment]
tl;dr: everything works well.

----

Thanks for this quick improvement and sorry for answering so late. Some changes from today made my initial draft for this answer obsolete ...

&nbsp;

I mainly tested with these settings 

    PRIORITIZE_TODAY = False
    IMPOSE_SUBDECK_LIMIT = False
    CUSTOM_SORT = SHOW_YOUNG_FIRST

Everything works. Small intervals are shown first. For cards with the same intervals (all cards with ivl = 1) the cards are shown in the order of subdecks and are not randomized between subdecks. This is exactly the same behavior as the add-on "change order of review cards" so I can replace the "Change Order ..." add-on (while using your enhanced options). Again, thank you very much for this add-on !! 

The new structure of your add-on makes it much easier to further modify it. I could even quickly implement a rough draft of my dream scheduler I mentioned in the other thread: more mature cards are fully randomized while cards with short intervals are sorted in ascending order and (contrary to your version or the add-on "Change Order ...") are randomized among subdecks. In case you're interested: https://github.com/ijgnd/HoochieMama/commit/9201cf312fc6ebb8e9fbe56fefe86a035547127d . Surprisingly I didn't notice a slowdown on my 6year old main machine. I'm curious how this change will fare on my old tablet with an Atom.
